src/components/Navbar.tsx

```diff
@@ -0,0 +1,228 @@
+  -moz-transition: 0.5s ease-in-out;
+  -webkit-transition: 0.5s ease-in-out;
+  transition: 0.5s ease-in-out;
+
+  &::before,
+  &::after {
+    -moz-transition: 0.5s ease-in-out;
+    -webkit-transition: 0.5s ease-in-out;
+    transition: 0.5s ease-in-out;
+    content: '';
+    position: absolute;
+    display: block;
+    width: 100%;
+    height: 100%;
+    background: ${({ theme }) => theme.color.primary};
+  }
+
+  &::before {
+    top: -10px;
+  }
+  &::after {
+    bottom: -10px;
+  }
+`;
+
+const fadeIn = keyframes`
+  from {
+    transform: scale(0.3);
+    opacity: 0;
+  }
+
+  to {
+    transform: scale(1);
+    opacity: 1;
+  }
+`;
+
+const fadeOut = keyframes`
+  from {
+    transform: scale(1);
+    opacity: 1;
+  }
+
+  to {
+    transform: scale(0.3);
+    opacity: 0;
+  }
+`;
+
+const LinkContainer = styled.div<{ menuOpen: boolean }>`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+  z-index: 200;
+  position: absolute;
+  top: 6rem;
+  right: 1rem;
+  @media ${({ theme }) => theme.device.tablet} {
+    right: 6rem;
+  }
+ 
+  visibility: ${(props) => (props.menuOpen ? 'visible' : 'hidden')};
+  -moz-transform-origin: top right;
+  -webkit-transform-origin: top right;
+  transform-origin: top right;
+  -moz-animation: ${(props) => (props.menuOpen ? fadeIn : fadeOut)} 0.25s
+  -webkit-animation: ${(props) =>
+    props.menuOpen ? fadeIn : fadeOut} 0.25s ease-in-out;
+  animation: ${(props) =>
+    props.menuOpen ? fadeIn : fadeOut} 0.25s ease-in-out;
+  -moz-transition: visibility 0.25s ease-in-out;
+  -webkit-transition: visibility 0.25s ease-in-out;
+  transition: visibility 0.25s ease-in-out;
+
+  a {
+    font-size: ${({ theme }) => theme.font.size.xxlarge};
+    -webkit-text-decoration-thickness: 4px;
+    text-decoration-thickness: 4px;
+    margin-bottom: 1rem;
+    display: block;
+    max-width: 100vw;
+
+    @media ${({ theme }) => theme.device.tablet} {
+    font-size: ${({ theme }) => theme.font.size.xxxlarge};
+    }
+  }
+`;
+
+const StyledInput = styled.input`
+  display: none;
+  -webkit-cursor: url(${Pointer}), pointer;
+  cursor: url(${Pointer}), pointer;
+
+  &:checked + ${Menu} {
+    box-shadow: 0 0 0 100vw ${({ theme }) => theme.color.background},
+      0 0 0 100vh ${({ theme }) => theme.color.background};
+    border-radius: 0;
+  }
+
+  &:checked + ${Menu} ${Hamburger} {
+    -moz-transform: rotate(45deg);
+    -webkit-transform: rotate(45deg);
+    transform: rotate(45deg);
+  }
+
+  &:checked + ${Menu} ${Hamburger}::after {
+    -moz-transform: rotate(90deg);
+    -webkit-transform: rotate(90deg);
+    transform: rotate(90deg);
+    bottom: 0;
+  }
+
+  &:checked + ${Menu} ${Hamburger}::before {
+    -moz-transform: rotate(90deg);
+    -webkit-transform: rotate(90deg);
+    transform: rotate(90deg);
+    top: 0;
+  }
+
+  &:checked + ${Menu} + ${LinkContainer} {
+    opacity: 1;
+  }
+`;
+
+interface NavProps {
+  menuOpen: boolean;
+}
+
+const Navbar: FC<NavProps> = () => {
```

Comment:
You should not use FC any more as this will be deprecated. FC has an implicit children prop which may not be desired or used. in this case you aren't using the children prop. The recommended approach would be to explicitly declare the children prop in your interface if you need it and pass it into VFC.
```suggestion
const Navbar: VFC<NavProps> = ({ menuOpen }) => {
```

side note: you aren't using the `menuOpen` prop in your interface or have destructured it in your params. Passing this interface to your `Navbar` component looks like it's not necessary.

Reply:
Removed FC, thanks :)

The menuOpen prop was for computing styles in css-in-js; however I've refactored and hoisted the hook up to the Header component to simplify the logic.